e2e_test/bin/e2e_test.dart

```diff
@@ -17,10 +17,26 @@ void main(List<String> arguments) async {
     print('AUTH TOKEN is not set');
     exit(1);
   }
+  SentryOptions? _options;
   await Sentry.init((options) {
     options.dsn = _exampleDsn;
+    _options = options;
   });
 
+  // We check if the configuration which can be set via environment or Dart
+  // defines are passed correctly in the application.
+  // The options callback has precedence over the environment, so we need to
+  // check those after init.
+  if (_options?.environment != 'e2e') {
+    print('Environment is not e2e');
+    exit(1);
+  }
+
+  if (_options?.dist != '1') {
+    print('Dist is not 1');
+    exit(1);
+  }
```

Comment:
the other option would be to `SentryEvent.fromJson` with the API response and assert the fields of the event, eg `event.dist`, `event.environment`, etc..., what do you think?

Reply:
That are two tests for the price of one :D

Reply:
indeed, but the idea is that we'd be expanding the assertion as discussed here https://github.com/getsentry/sentry-dart/issues/514#issuecomment-905830314
so asserting `environment` and `dist` would not be a problem, your call, the current solution also works, but holding the reference of `_options` is also very hacky :)